operator/redisfailover/service/generator.go

```diff
@@ -655,7 +655,7 @@ func generateRedisStatefulSet(rf *redisfailoverv1.RedisFailover, labels map[stri
 			ServiceName: name,
 			Replicas:    &rf.Spec.Redis.Replicas,
 			UpdateStrategy: appsv1.StatefulSetUpdateStrategy{
-				Type: appsv1.OnDeleteStatefulSetStrategyType,
+				Type: appsv1.RollingUpdateStatefulSetStrategyType,
```

Comment:
Could this be dangerous? As I understand it, STSs usually use a deletion strategy to prevent the situation where multiple versions of an application could be running at the same time. 

> When a StatefulSet's .spec.updateStrategy.type is set to RollingUpdate, the StatefulSet controller will delete and recreate each Pod in the StatefulSet. It will proceed in the same order as Pod termination (from the largest ordinal to the smallest), updating each Pod one at a time.
>
> The Kubernetes control plane waits until an updated Pod is Running and Ready prior to updating its predecessor. -- https://kubernetes.io/docs/concepts/workloads/controllers/statefulset/#rolling-updates

The assumption being that disparate versions of the stateful application sharing data could lead to corruption. 

> When using [Rolling Updates](https://kubernetes.io/docs/concepts/workloads/controllers/statefulset/#rolling-updates) with the default [Pod Management Policy](https://kubernetes.io/docs/concepts/workloads/controllers/statefulset/#pod-management-policies) (OrderedReady), it's possible to get into a broken state that requires [manual intervention to repair](https://kubernetes.io/docs/concepts/workloads/controllers/statefulset/#forced-rollback) -- https://kubernetes.io/docs/concepts/workloads/controllers/statefulset/#limitations

Reply:
pod management policy is set to [parallel](https://github.com/powerhome/redis-operator/blob/8062906e5450fe74200f6028951910c60f501f7d/operator/redisfailover/service/generator.go#L660), but it is does not matter during updates:

> Parallel pod management tells the StatefulSet controller to launch or terminate all Pods in parallel, and to not wait for Pods to become Running and Ready or completely terminated prior to launching or terminating another Pod. This option only affects the behavior for scaling operations. Updates are not affected.